python_modules/dagster/dagster_tests/definitions_tests/declarative_automation_tests/automation_condition_tests/builtins/test_check_status_condition.py

```diff
@@ -0,0 +1,123 @@
+import pytest
+from dagster import (
+    AssetCheckKey,
+    AssetCheckResult,
+    AssetCheckSpec,
+    AssetExecutionContext,
+    AssetKey,
+    AutomationCondition,
+    Definitions,
+    Output,
+    asset,
+    asset_check,
+    evaluate_automation_conditions,
+    instance_for_test,
+)
+from dagster._core.storage.asset_check_execution_record import (
+    AssetCheckExecutionResolvedStatus as ACS,
+)
+
+
+def _get_defs(status: ACS, automation_condition: AutomationCondition) -> Definitions:
+    @asset
+    def a() -> None: ...
+
+    @asset_check(asset=a, automation_condition=automation_condition)
+    def some_check() -> AssetCheckResult:
+        if status == ACS.SUCCEEDED:
+            return AssetCheckResult(passed=True)
+        elif status == ACS.FAILED:
+            return AssetCheckResult(passed=False)
+        elif status == ACS.EXECUTION_FAILED:
+            raise Exception("fail")
+        else:
+            raise Exception(f"unexpected status {status}")
+
+    return Definitions(assets=[a], asset_checks=[some_check])
+
+
+@pytest.mark.parametrize(
+    "status, automation_condition",
+    [
+        (ACS.SUCCEEDED, AutomationCondition.check.succeeded()),
+        (ACS.FAILED, AutomationCondition.check.failed()),
+        (ACS.EXECUTION_FAILED, AutomationCondition.check.execution_failed()),
+    ],
+)
+def test_status_conditions(status: ACS, automation_condition: AutomationCondition) -> None:
+    defs = _get_defs(status, automation_condition)
+    asset_job = defs.get_implicit_global_asset_job_def()
```

Comment:
IMO these type of. parameterizations often hurt more than they help as they are hard to parse. Would rather seem some more straightforward examples that more clearly mimic what the users would write.

Reply:
I generally agree, and I think the last test here does a better job of illustrating a "real-world" use case (at the expense of being less targeted).

There's a bit of inherent weirdness here with testing a single automation condition in a vacuum, as many of these "basic" automation conditions would not make sense as a standalone condition on an asset.

I'd prefer to keep this specific test around (as it does capture some valuable cases), but I also added an additional "more realistic" test underneath.